### PR TITLE
[speech-commands] Add fine-tuning and demo

### PR DESCRIPTION
**Pull request by caisq**
- Fine-tuning happens after the initial transfer training and unfreezes a Dense layer in the base model
- Demo: require a minimum number of examples per category
- Demo: Allow fine-tuning and plot loss and accuracy values during fine-tuning
- Add util for balanced train-validation splits

This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/92)

### CODE REVIEW

**Comment by giapseng** [comment]
Can teach me create a ai teaching machine app ?

**Comment by caisq** [comment]
Gentle ping @pyu10055

**Comment by caisq** [comment]
The state of this PR is messed up. I'll open a new PR.
